kratos/sources/model_part.cpp

```diff
@@ -855,6 +855,293 @@ void ModelPart::RemoveElementsFromAllLevels(Flags identifier_flag)
+                                                                                        rMasterVariable,
+                                                                                        rSlaveNode,
+                                                                                        rSlaveVariable,
+                                                                                        Weight,
+                                                                                        Constant);
+
+    mMasterSlaveConstraints.push_back(p_new_constraint);
+    mMasterSlaveConstraints.Unique();
+
+    return p_new_constraint;
+    KRATOS_CATCH("")
+
+}
+
+
+/** Remove the master-slave constraint with given Id from mesh with ThisIndex in this modelpart and all its subs.
+*/
+void ModelPart::RemoveMasterSlaveConstraint(ModelPart::IndexType MasterSlaveConstraintId)
+{
+    mMasterSlaveConstraints.erase(MasterSlaveConstraintId);
+
+    for (SubModelPartIterator i_sub_model_part = SubModelPartsBegin(); i_sub_model_part != SubModelPartsEnd(); i_sub_model_part++)
+        i_sub_model_part->RemoveMasterSlaveConstraint(MasterSlaveConstraintId); 
+}
+
+/** Remove given master-slave constraint from mesh with ThisIndex in this modelpart and all its subs.
+*/
+void ModelPart::RemoveMasterSlaveConstraint(ModelPart::MasterSlaveConstraintType& ThisMasterSlaveConstraint)
+{
+    this->RemoveMasterSlaveConstraint(ThisMasterSlaveConstraint.Id());
+    for (SubModelPartIterator i_sub_model_part = SubModelPartsBegin(); i_sub_model_part != SubModelPartsEnd(); i_sub_model_part++)
+        i_sub_model_part->RemoveMasterSlaveConstraint(ThisMasterSlaveConstraint.Id());
+}
+
+
+/** Remove the master-slave constraint with given Id from mesh with ThisIndex in parents, itself and children.
+*/
+void ModelPart::RemoveMasterSlaveConstraintFromAllLevels(ModelPart::IndexType MasterSlaveConstraintId)
+{
+
+    if (IsSubModelPart())
+        mpParentModelPart->RemoveMasterSlaveConstraint(MasterSlaveConstraintId);
+    else
+        RemoveMasterSlaveConstraint(MasterSlaveConstraintId);
+
+}
+
+/** Remove given master-slave constraint from mesh with ThisIndex in parents, itself and children.
+*/
+void ModelPart::RemoveMasterSlaveConstraintFromAllLevels(ModelPart::MasterSlaveConstraintType& ThisMasterSlaveConstraint)
+{
+
+    if (IsSubModelPart())
+        mpParentModelPart->RemoveMasterSlaveConstraint(ThisMasterSlaveConstraint);
+    else
+        RemoveMasterSlaveConstraint(ThisMasterSlaveConstraint);
+
+}
+
+/** Returns the MasterSlaveConstraint::Pointer  corresponding to it's identifier */
+ModelPart::MasterSlaveConstraintType::Pointer ModelPart::pGetMasterSlaveConstraint(ModelPart::IndexType MasterSlaveConstraintId)
+{
+        auto i = mMasterSlaveConstraints.find(MasterSlaveConstraintId);
+        KRATOS_DEBUG_ERROR_IF(i == mMasterSlaveConstraints.end()) << " constraint index not found: " << MasterSlaveConstraintId << ".";
```

Comment:
Should this be a debug error only?
The effects of this being unnoticed are quite severe I'd say => consider doing this check always
(I haven't checked how it is done for elems/ conds)

Reply:
You are correct ... It should be done always. I change it.

Reply:
let me disagree on this. It is like the [] operator on a std vector. the check is only done in debug...